src/web3/WalletClient.ts

```diff
@@ -328,31 +313,34 @@ export class WalletClient extends BaseClient implements IWalletClient {
 
   /**
    * Generates a new wallet account.
+   * @param version_number - The version number of the secret key to be generated, to create a new account.
    *
    * @returns A Promise that resolves to an {@link IAccount} object, which represents the newly created account.
    */
-  public static async walletGenerateNewAccount(): Promise<IAccount> {
+  public static async walletGenerateNewAccount(
+    version_number: number,
```

Comment:
version number in arg to generate a new Account. Moreover, it is here the secret key version, and not the public key or address version